### ISSUE TITLE
Errors emitted for unrecognized reST directives outside the robotframework code block, introduced in v4.1.2

### ISSUE THREAD

**Issue by alangefe** [robotframework/robotframework#4124]
# Bug description

`robot` is now emitting errors for unrecognized reST directives beginning in v4.1.2, even though the directives are not inside the `robotframework` code block. The errors do not cause the `robot` process to fail, but they do clutter up the console.

This issue was introduced by the changes in #4085 . In v4.1.1, these errors did not appear.

# Reproduction

Use the following sample reST file:

```
.. highlight:: robotframework

===========
Sample Test
===========

.. contents::
    :local:

Overview
========

This is a sample test case.

Tests
=====

.. code:: robotframework

    *** Test Cases ***
    Sample Test Case
        No Operation
```

Run it using `robot sample.rst`.

## Expected output

```
==============================================================================
Sample                                                                        
==============================================================================
Sample Test Case                                                      | PASS |
------------------------------------------------------------------------------
Sample                                                                | PASS |
1 test, 1 passed, 0 failed
==============================================================================
Output:  <path>/output.xml
Log:     <path>/log.html
Report:  <path>/report.html
```

## Actual output

```
<path>/sample.rst:1: (ERROR/3) Error in "highlight" directive:
no content permitted.

.. highlight:: robotframework

<path>/sample.rst:7: (ERROR/3) Error in "contents" directive:
no content permitted.

.. contents::
    :local:

==============================================================================
Sample                                                                        
==============================================================================
Sample Test Case                                                      | PASS |
------------------------------------------------------------------------------
Sample                                                                | PASS |
1 test, 1 passed, 0 failed
==============================================================================
Output:  <path>/output.xml
Log:     <path>/log.html
Report:  <path>/report.html
```

# Version tested
Robot Framework 4.1.2 (Python 3.7.10 on linux)
docutils 0.17.1

**Comment by pekkaklarck** [comment]
This is very unfortunate. The new approach to ignore errors when parsing reST files introduced in #4085 (issue #4086) catches some errors than the original approach introduced in #2152 (issue #2093) didn't catch but very unfortunately misses some others. 

This definitely needs to be fixed. I was hoping RF 4.1.2 would be the last RF 4 release, but I consider this annoying enough regression that it warrants RF 4.1.3 release. How urgently do you need a fix? I'd like to wait some time to see are there other bug reports, especially regressions, that we should fix as well.

**Comment by alangefe** [comment]
This is not an urgent issue for me, since it is mostly cosmetic.

**Comment by pekkaklarck** [comment]
RF 4.1.3 has now been scheduled for December 15. That's exactly two months after RF 4.1.2.

**Comment by pekkaklarck** [comment]
The earlier fix has now been committed also to 4.1 maintenance branch. I'll create release shortly.
